src/Juvix/Backends/Michelson/Parameterisation.hs

```diff
@@ -37,6 +44,52 @@ constType v =
 arity :: PrimVal -> Int
 arity = pred . length . typeOf
 
+applyProper ::
+  Prim.Take PrimTy PrimVal ->
+  [Prim.Take PrimTy PrimVal] ->
+  Either
+    (CoreTypes.PipelineError PrimTy PrimVal)
+    (Prim.Return PrimTy PrimVal)
+applyProper fun args =
+  case Prim.term fun of
+    Constant _i ->
+      case length args of
+        0 ->
+          Right (Prim.Return (Prim.term fun))
+        _x ->
+          Left (CoreTypes.TypecheckerError "Applied a constant to argument")
+    Inst instruction ->
```

Comment:
Somewhere we need to deal with dependency, namely, polymorphic functions in Michelson will be dependent in Juvix, e.g. `ADD`:

```idris
add : (t : Type) -> t -> t -> t
```

— so we need to deal with the first application to `t` (e.g. `Int` or `Tez`) somewhere, how do you think we should do that?